charts/system/kubelet-csr-approver/templates/_servicemonitor.tpl

```diff
@@ -0,0 +1,29 @@
+{{- define "csr.servicemonitor" -}}
+{{- if hasKey .Values "metrics" }}
+{{- if .Values.metrics.enabled }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "tc.common.names.fullname" . }}
+  labels:
+    {{- include "tc.common.labels" . | nindent 4 }}
+    {{- with .Values.metrics.serviceMonitor.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "tc.common.labels.selectorLabels" . | nindent 6 }}
+  endpoints:
+    - port: metrics
+      {{- with .Values.metrics.serviceMonitor.interval }}
+      interval: {{ . }}
+      {{- end }}
+      {{- with .Values.metrics.serviceMonitor.scrapeTimeout }}
+      scrapeTimeout: {{ . }}
+      {{- end }}
+      path: /metrics
+{{- end }}
+{{- end }}
+{{- end -}}
```

Comment:
Use the metrics section, don't start creating DIY objects that are already handled with common...